src/layouts/Catalogue/Catalogue.js

```diff
@@ -1,358 +1,149 @@
-
-  addtoCartHandler = () => {
-    var carts = this.state.cart
-    var qty = this.state.qty
+  function addtoCartHandler() {
+    var carts = stateCart.cart;
+    var qty = state.quantity;
     var position = carts
-    .map(function (e) {
-      return e.id;
-    })
-    .indexOf(this.state.selectedProduct.id);
-    if (position<0){
+      .map(function (e) {
+        return e.id;
+      })
+      .indexOf(state.selectedProduct.id);
+    if (position < 0) {
       //new insert
       var objInserted = {
-        ...this.state.selectedProduct,
+        ...state.selectedProduct,
         qty,
       };
-      carts.push(objInserted)
-    }else{
+      carts.push(objInserted);
+    } else {
       var obj = carts[position];
-      var objInserted = {...obj, qty: obj.qty+this.state.qty};
+      var objInserted = { ...obj, qty: obj.qty + state.quantity };
       carts[position] = objInserted;
     }
-    this.setState({
-      showModal: false,
-      qty: 1,
-      selectedProduct: {},
-    },()=>{
-      this.recalculateGrandTotal();
-      console.log(`cart`, this.state.cart)
-    })
-  }
-
-  handlerEmptyCart(){
-    this.setState({
-      cart: [],
-    })
+    actions.reset();
+    recalculateGrandTotal();
   }
 
-  renderModalCart(){
-    const Context = styled.div`
-       display: flex;
-       flex-direction: row;
-       margin: 30px;
-    `;
-
-    const ItemInfo = styled.div`
-      display: flex;
-      flex-direction: row;
-    `;
-
-    const Image = styled.img`
-      height: 150px;    
-    `;
+  function recalculateGrandTotal() {
+    var grandTotal = 0.0,
+      itemTotal = 0;
 
-    const ItemDetail = styled.div`
-      margin: 20px;
-      padding: 10px;
-    `;
-
-    const GrandTotal = styled.div`
-      display: flex;
-      align-items: flex-end;
-      justify-content: flex-end; 
-    `;
-
-    return(
-      <Modal
-      centered
-      title="Cart"
-      visible={this.state.showModalCart}
-      onOk={this.handleOkCart}
-      onCancel={this.handleCancelCart}
-      >
-      {(this.state.cart.length == 0)?
-        <p>Your cart is Empty</p>:
-        <div>
-          {this.state.cart.map((product)=>{
-            return(
-              <Context>
-                <ItemInfo>
-                  <Image src={product.image}/>
-                  <ItemDetail>
-                    <h2>{product.title}</h2>
-                    <strong>{product.qty} x ${product.price}</strong><br/>
-                  </ItemDetail>
-                </ItemInfo>
-                <GrandTotal>
-                  ${product.qty*product.price}
-                </GrandTotal>
-              </Context>
-            )
-          })
-        }
-        <Button onClick={()=>{this.handlerEmptyCart()}}>EmptyCart</Button>
-        </div>
-        }
-    </Modal>
-    )
-  } 
-
-  recalculateGrandTotal = () => {
-    var grandTotal = 0.0, itemTotal = 0;
-    
-    this.state.cart.map((v) => {
-      grandTotal += v.qty*v.price;
+    stateCart.cart.map((v) => {
```

Comment:
What does `v` mean? I think it's an item in the product cart, so I should call it a `product` instead.